### DISCUSSION TITLE
automatic startup on openwrt 22.03.3 on Linux_arm64

### DISCUSSION THREAD

**Discussion by Tonho888** [0xERR0R/blocky#827] [Q&A] [answered]
I'm tired of trying
I have the config in /opt/blocky --config /opt/config.yml, and it runs normally but fails to startup when the router is restarted,sir
and when the command is run again via putty, everything returns to normal

how to make automatic startup blocky on openwrt 22.03.3 Linux_arm64 sir, please help me

**Comment by 0xERR0R** [comment]
Maybe this is a permission problem? If you start blocky via ssh, it runs under your user. Wich user is used if it starts automatically? Does this user have read permissions to read the config file? Maybe you can try to grant read permissions to all groups.

**Comment by Tonho888** [comment, marked as answer]
after trying and failing it finally succeeded, I put rc.local "/opt/blocky --config /opt/config.yml" before exit0, and restarted, and it ran perfectly, but I lost the restart button on the router.just unplug power and plug in, blocky runs again. thank you sir
